File: JavaKotlinDers9ParsLearning/app/src/main/java/com/example/javakotlinders9parslearning/MainActivity.java
package com.example.javakotlinders9parslearning;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Toast;

import com.parse.FindCallback;
import com.parse.LogInCallback;
import com.parse.ParseException;
import com.parse.ParseObject;
import com.parse.ParseQuery;
import com.parse.ParseUser;
import com.parse.SaveCallback;
import com.parse.SignUpCallback;

import java.util.List;

/*
* pars kullanabilmek için build:gradle appimize
ext {
    parseVersion = "1.17.3"
}
dependencies{
implementation "com.parse:parse-android:$parseVersion"
    implementation "com.parse:parse-fcm-android:$parseVersion"
}  eklemelerimizi yapıyoruz.
*parse'ı projemizde başlatabilmemiz için yeni bir sınıf oluşturmamız gerekiyor.
ayrı bir sınıfta işlem yapmamız gerekiyor.
*Pars server'ı ile nasıl çalışılır?
back4app e giriyoruz internetten. Parse server sitesi bu site
buil new app-server settings-core settings-ordaki verileri parseStarter sınıfına ekliyoruz.
server=parse api address demek. bağlantımızı yapıyoruz.
*manage parse settingte parse versiyonun yani server seçimin var. Betalar önerilmiyor.
*artık parsa veri kaydetmeye hazırız.
*
*
*
*
* */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        kullaniciylaGirisYap();

    }

    public void veritabaninaVeriKaydet() {
        ParseObject object = new ParseObject("Fruits");
        /*VERİ KAYDETME*/
        object.put("name", "apple");
        object.put("calories", 100);
        //savenİnBackground'un callbackli olanınında işlem başarılı olursa ya da başarısız olursa gibi seçenekleri de mevcut.
        object.saveInBackground(new SaveCallback() {
            @Override
            public void done(ParseException e) {
                if (e != null) {
                    Toast.makeText(getApplicationContext(), e.getLocalizedMessage(), Toast.LENGTH_LONG).show();

                }
            }
        });
    }

    public void veritabanindanVeriCek() {
        /*VERİ OKUMA*/
        ParseQuery<ParseObject> query = ParseQuery.getQuery("Fruits");
        query.findInBackground(new FindCallback<ParseObject>() {
            @Override
            public void done(List<ParseObject> objects, ParseException e) {
                if (e != null) {
                    e.printStackTrace();
                } else {
                    if (objects.size() > 0) {
                        for (ParseObject object : objects) {
                            String objectName = object.getString("name");
                            int objectCalories = object.getInt("calories");
                            System.out.println(objectName);
                            System.out.println(objectCalories);

                        }
                    }
                }
            }
        });
        /*filtreleme de yapabiliriz. Mesela sadece muzu çekmek istiyoruz diyelim
        query.whereEqualTo("name", "apple");
        query.findInBackground();......aynısı yapınca sadece elma olanlar gelicek.*/

    }

    public void kullaniciOlustur() {
        ParseUser user = new ParseUser();//diyip bir kullanıcı oluşturabiliyoruz.
        user.setUsername("James");
        user.setPassword("123456");

        user.signUpInBackground(new SignUpCallback() {
            @Override
            public void done(ParseException e) {
                if (e != null) {
                    e.printStackTrace();
                } else {
                    Toast.makeText(getApplicationContext(), "kullanici olusturuldu", Toast.LENGTH_LONG).show();
                }
            }
        });

    }

    public void kullaniciylaGirisYap() {
        ParseUser.logInInBackground("James", "123456", new LogInCallback() {
            @Override
            public void done(ParseUser user, ParseException e) {
                if (e != null) {
                    e.printStackTrace();
                } else {
                    Toast.makeText(getApplicationContext(), "Hosgeldin" + user.getUsername(), Toast.LENGTH_LONG).show();

                }
            }
        });


    }


}

File: KotlinDers16FourSquareParse/app/src/main/java/com/example/kotlinders16foursquareparse/MainActivity.kt
package com.example.kotlinders16foursquareparse

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.parse.ParseObject
import com.parse.ParseQuery
import com.parse.ParseUser
import kotlinx.android.synthetic.main.activity_main.*

/**
 * gradle eklemelerini yap
 * parse ile sunucumuzu bağlamak için bir sınıf oluşturacağız
 *
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //veriKaydet()
        //veriOku()
        //filtreliVeriCek()
        kontrol()
        action()

    }

    fun veriKaydet() {
        //parse'a veri kaydetmek için
        val parseObj =
            ParseObject("Fruits")  //fruits diye bir sınıf oluşturduk(varsa bi daha oluşturmuyo ekleme yapıyo)
        parseObj.put("name", "muz")  //bu verileri ekledik
        parseObj.put("calories", 100)
        //callback'li bişeyi seçmek demek yaptığı işlemin sonucunu da verecek demektir
        parseObj.saveInBackground { e ->
            if (e != null) {
                Toast.makeText(applicationContext, e.localizedMessage.toString(), Toast.LENGTH_LONG)
                    .show()
            } else {
                Toast.makeText(applicationContext, "veri kaydoldu", Toast.LENGTH_LONG).show()

            }
        }
    }

    fun veriOku() {
        //parse'dan veri çekelim
        val query = ParseQuery.getQuery<ParseObject>("Fruits")
        query.findInBackground { objects, e ->
            if (e != null) {
                Toast.makeText(applicationContext, e.localizedMessage.toString(), Toast.LENGTH_LONG)
                    .show()
            } else {
                if (objects.size > 0) {
                    for (parseObject in objects) {
                        val name = parseObject.get("name") as String
                        val calories = parseObject.get("calories") as Int
                        println("name: $name")
                        println("calories: $calories")
                    }
                }
            }
        }

    }

    fun filtreliVeriCek() {
        //filtreli bir şekilde veri çekmek istersem eğer:
        val query2 = ParseQuery.getQuery<ParseObject>("Fruits")
        query2.whereEqualTo("name", "muz").findInBackground { objects, e ->
            for (parseObject in objects) {
                val name = parseObject.get("name") as String
                val calories = parseObject.get("calories") as Int
                println("name: $name")
                println("calories: $calories")
            }
        }

        val query3 = ParseQuery.getQuery<ParseObject>("Fruits")
        query3.whereLessThan("calories", 300)//kalorisi 300den az olanları getir dedik
        query3.findInBackground { objects, e ->
            for (parseObject in objects) {
                val name = parseObject.get("name") as String
                val calories = parseObject.get("calories") as Int
                println("name: $name")
                println("calories: $calories")
            }
        }
    }

    fun action(){
        signUpButton.setOnClickListener {
            val user=ParseUser()
            user.username=userNameEditText.text.toString()
            user.setPassword(passwordEditText.text.toString())
            user.signUpInBackground {e ->
                if (e!=null){
                    Toast.makeText(applicationContext, e.localizedMessage.toString(), Toast.LENGTH_LONG).show()

                }else{
                    Toast.makeText(applicationContext, "User Created", Toast.LENGTH_LONG).show()
                    val intent=Intent(applicationContext,LocationsActivity::class.java)
                    startActivity(intent)
                    finish()
                }
            }

        }

        signInButton.setOnClickListener {
            ParseUser.logInInBackground(userNameEditText.text.toString(),passwordEditText.text.toString()){user, e ->
                if (e!=null){
                    Toast.makeText(applicationContext, e.localizedMessage, Toast.LENGTH_LONG).show()

                }else{
                    Toast.makeText(applicationContext, "Hosgeldin: ${user.username}", Toast.LENGTH_LONG).show()
                    val intent=Intent(applicationContext,LocationsActivity::class.java)
                    startActivity(intent)
                    finish()
                }
            }
        }
    }

    fun kontrol(){
        val user=ParseUser.getCurrentUser()
        if(user.username!=null){
            val intent=Intent(applicationContext,LocationsActivity::class.java)
            startActivity(intent)
            finish()
        }
    }

}
